abipy/abio/factories.py

```diff
@@ -417,103 +417,200 @@ def g0w0_with_ppmodel_inputs(structure, pseudos,
+    #if gamma:
+    #    scf_ksampling = KSampling.automatic_density(structure=structure, kppa=10000, chksymbreak=0, shifts=(0, 0, 0))
+    #    nscf_ksampling = KSampling.gamma_centered(kpts=(2, 2, 2))
+    #    if kppa <= 13:
+    #        nscf_ksampling = KSampling.gamma_centered(kpts=(scf_kppa, scf_kppa, scf_kppa))
+    #    else:
+    #        nscf_ksampling = KSampling.automatic_density(structure, scf_kppa, chksymbreak=0, shifts=(0, 0, 0))
+    #else:
+    #    scf_ksampling = KSampling.automatic_density(structure, scf_kppa, chksymbreak=0)
+    #    nscf_ksampling = KSampling.automatic_density(structure, scf_kppa, chksymbreak=0)
+
+    if gamma:
+        if kppa == 1:
+            scf_ksampling = aobj.KSampling.gamma_centered(kpts=(1, 1, 1))
+            nscf_ksampling = aobj.KSampling.gamma_centered(kpts=(1, 1, 1))
+        elif kppa == 2:
+            scf_ksampling = aobj.KSampling.gamma_centered(kpts=(2, 2, 2))
+            nscf_ksampling = aobj.KSampling.gamma_centered(kpts=(2, 2, 2))
+        elif kppa < 0:
+            scf_ksampling = aobj.KSampling.gamma_centered(kpts=(-kppa, -kppa, -kppa))
+            nscf_ksampling = aobj.KSampling.gamma_centered(kpts=(2, 2, 2))
+        elif kppa <= 13: 
+            scf_ksampling = aobj.KSampling.gamma_centered(kpts=(kppa, kppa, kppa))
+            nscf_ksampling = aobj.KSampling.gamma_centered(kpts=(kppa, kppa, kppa))
+        else:
+            scf_ksampling = aobj.KSampling.automatic_density(structure, kppa, chksymbreak=0, shifts=(0, 0, 0))
+            nscf_ksampling = aobj.KSampling.automatic_density(structure, kppa, chksymbreak=0, shifts=(0, 0, 0))
+    else:
+        # this is the original behaviour before the devellopment of the gwwrapper
+        scf_ksampling = KSampling.automatic_density(structure, kppa, chksymbreak=0)
+        nscf_ksampling = KSampling.automatic_density(structure, kppa, chksymbreak=0)
 
-    multi_scf = MultiDataset(structure, pseudos, ndtset=len(scf_diffs))
+    scf_electrons = aobj.Electrons(spin_mode=spin_mode, smearing=smearing, algorithm=scf_algorithm,
+                                   charge=charge, nband=scf_nband, fband=None)
+    nscf_electrons = aobj.Electrons(spin_mode=spin_mode, smearing=smearing, algorithm={"iscf": -2},
+                                    charge=charge, nband=nscf_nband, fband=None)
+
+    multi_scf = MultiDataset(structure, pseudos, ndtset=max(1, len(scf_diffs)))
    
     multi_scf.set_vars(scf_ksampling.to_abivars())
     multi_scf.set_vars(scf_electrons.to_abivars())
+    multi_scf.set_vars(extra_abivars_all)
+    multi_scf.set_vars(_stopping_criterion(runlevel="scf", accuracy=accuracy))
     multi_scf.set_vars(extra_abivars)
-    for vars, abinput in zip(scf_diffs, multi_scf):
-	print(type(vars), type(abinput))
-        abinput.set_vars(vars)   
- 
-#multi_scf.set_vars(smearing.to_abivars())
 
-    multis.append(multi_scf)
-
-    return multis
+    for vars, abinput in zip(scf_diffs, multi_scf):
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Redefining built-in 'vars'](https://www.codacy.com/app/gmatteo/abipy/file/2694193837/issues/source?bid=3176346&fileBranchId=3340437#l543)